heap/heap.py

```diff
@@ -3,19 +3,56 @@ def __init__(self):
     self.storage = []
 
   def insert(self, value):
-    pass
+    self.storage.append(value)
+    indexOfValue = len(self.storage) -1
+    while(True):
+      indexOfValue = self._bubble_up(indexOfValue)
+      if indexOfValue is False:
+        break
 
   def delete(self):
-    pass
+    if self.get_size() == 1:
+      return self.storage.pop()
+    oldTop = self.storage[0]
+    self.storage[0]= self.storage.pop()
+    indexOfValue = 0
+    while(True):
+      indexOfValue = self._sift_down(indexOfValue)
+      if indexOfValue is False:
+        return oldTop
 
   def get_max(self):
-    pass
+    return self.storage[0]
 
   def get_size(self):
-    pass
+    return len(self.storage)
 
   def _bubble_up(self, index):
-    pass
+    if index == 0:
+      return False
+    child = self.storage[index]
+    parent = self.storage[(index - 1)//2]
+    if child > parent:
+      self.storage[index] = parent
+      self.storage[(index - 1)//2] = child 
+      return (index - 1)//2
+    return False
 
```

Comment:
If only I had thought to name my stuff parent and child I would have saved myself a lot of headaches. Nice!